### DISCUSSION TITLE
Cannot import osmium on Amazon Athena Notebook

### DISCUSSION THREAD

**Discussion by ruanqizhen** [osmcode/pyosmium#240] [Q&A]
I’m currently executing Python code on an Amazon Athena Notebook and attempting to utilize some methods from Osmium. However, I’m encountering difficulties importing osmium. I’ve been following the instructions provided [here by AWS](https://docs.aws.amazon.com/athena/latest/ug/notebooks-import-files-libraries.html#notebooks-import-files-libraries-adding-python-files), which essentially require me to use “pip install -t” to install PyOsmium in a separate folder. 

Despite these efforts, I’m unable to import it. The error message I consistently receive is:

```
__init__.py", line 8, in <module>
    from osmium._osmium import *
ModuleNotFoundError: No module named 'osmium._osmium'
```

Interestingly, I’ve tried importing other Python models for the specific folder and have been successful. I suspect the issue with Osmium might be due to it providing only .pyi files, not .py files. I’m uncertain as to why Osmium can be imported when installed in the default directory.

Does anyone have any suggestions on how I can successfully load the model? I appreciate your help!

**Comment by lonvia** [comment]
pyosmium contains compiles C++ code. You need the appropriate shared libraries (*.so files). If and how this works in Athena Notebooks, I don't know.

**Reply by ruanqizhen** [reply]
I installed the package by calling "pip install -t", appropriate *.so files are also installed

**Comment by ruanqizhen** [comment]
I found the problem.
Amazon Athena Notebook can only import a module from a zip file.  But in Python, ZIP import of dynamic modules (.pyd, .so) is disallowed.
